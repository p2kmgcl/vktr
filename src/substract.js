import add from "./add";
import haveSameNumberOfDimensions from "./assertions/have-same-number-of-dimensions";
import isVector from "./assertions/is-vector";
import multiply from "./multiply";

/**
 * Returns a new vector result of (a - b)
 * @param {Vector} a
 * @param {Vector} b
 * @return {Vector}
 */
function substract(a, b) {
  isVector(a);
  isVector(b);
  haveSameNumberOfDimensions(a, b);

  return add(a, multiply(b, -1));
}

export default substract;
