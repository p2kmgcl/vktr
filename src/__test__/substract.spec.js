import substract from "../substract";
import Vector from "../vector";

describe("substract()", () => {
  it("allows substracting two vectors", () => {
    const a = new Vector(1);
    const b = new Vector(2);

    expect(substract(a, b).components).toMatchSnapshot();
  });

  it("requires vector parameter", () => {
    const a = new Vector(1);

    expect(() => substract(a, 1)).toThrow(TypeError);
  });

  it("requires same number of dimensions", () => {
    const a = new Vector(1);
    const b = new Vector(1, 2);

    expect(() => substract(a, b)).toThrow(RangeError);
  });
});
