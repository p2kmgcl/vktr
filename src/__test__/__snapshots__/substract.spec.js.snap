// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`substract() allows substracting two vectors 1`] = `
Vector {
  "components": Array [
    -1,
  ],
}
`;
